standards/ICRC-3/README.md

```diff
@@ -0,0 +1,260 @@
+
+## Blocks Verification
+
+The Ledger MUST certify the last block (tip) recorded. The Ledger MUST allow to download the certificate via the `icrc3_get_tip_certificate` endpoint. The certificate follows the [IC Specification for Certificates](https://internetcomputer.org/docs/current/references/ic-interface-spec#certification). The certificate contains a tree with the certified data and the signature. The tree MUST contain two labelled values (leafs):
+1. `last_block_index`: the index of the last block in the chain. The values must be expressed as big-endian
+2. `last_block_hash`: the hash of the last block in the chain
+
+Clients SHOULD download the tip certificate first and then download the block backward starting from `last_block_index` and validate the blocks in the process.
+
+Validation of block `i` is done by checking the block hash against
+1. if `i + 1 < len(chain)` then the parent hash `phash` of the block `i+1`
+2. otherwise the `last_block_hash` in the tip certificate.
+
+## Interaction with other standards
+
+Each standard that adheres to `ICRC-3` must define the list of transactions types they define and/or extend together with their `Value` schema and the function that converts a [`Value`](#value) to that type. Transaction types are well-typed records that are easy to consume by clients.
+
+`Value`s representing blocks must have the following schema properties:
+1. they must be of type `Map`
+1. they must have a field `tx: Map` which describes the transaction inside the block
+1. they must have a field `op: Text` inside `tx` which describes the type of transactions, e.g. "ICRC1_Burn" for `ICRC1_Burn` transactions
+1. all blocks with `index > 0` must have a top-level field called `phash: Blob` which contains the [hash](#value-hash) of the previous block.
+
+In other words the schema must be:
+
+```
+type ICRC1_Block = {
+    // The hash of the parent Transaction
+    "phash" : Blob?;
+
+    // The timestamp of when the block was
+    // added to the Ledger
+    "ts": u64,
+
+    // The effective fee of the block. This is
+    // empty if the user specified a fee in the
+    // transaction tx, otherwise it contains the
+    // fee
+    "fee": Nat?,
+
+    // The transaction inside the block
+    "tx" : {
+        "op" : Text;
+        ...
+    };
+};
+```
+
+where the `...` in `"tx"` are the rest of the fields for the specific transaction.
+
+For instance, [`ICRC-1`](https://github.com/dfinity/ICRC-1/tree/main/standards/ICRC-1) should define three transactions types - `ICRC1_Mint`, `ICRC1_Burn` and `ICRC1_Transfer` - and the function to convert a `Value` to them in order to adhere to the `ICRC-3` standard.
+
+## Specification
+
+### `icrc3_get_blocks`
+
+```
+type Value = variant {
+    Blob : blob;
+    Text : text;
+    Nat : nat; // do we need this or can we just use Int?
+    Int : int;
+    Array : vec Value;
+    Map : vec record { text; Value };
+};
+
+type GetBlocksArgs = vec record { start : nat; length : nat };
```

Comment:
How do I read _the latest_ block?
This is crucial for any block explorer.

Reply:
`icrc3_get_tip_certificate` returns the latest block number together with its hash (both values are certified).